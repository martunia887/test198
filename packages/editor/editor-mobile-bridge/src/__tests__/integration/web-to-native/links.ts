import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';

import {
  callNativeBridge,
  editor,
  editable,
  skipBrowsers as skip,
  getBridgeOutput,
} from '../_utils';

// https://product-fabric.atlassian.net/browse/ED-6877
BrowserTestCase(
  'currentSelection when no selection',
  // Safari has issues with key events
  { skip: skip.concat('safari', 'chrome') },
  async (client: any, testName: string) => {
    const browser = new Page(client);
    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await browser.type(editable, 'Normal Text');
    await browser.keys('ArrowLeft');

    const currentSelection = await getBridgeOutput(
      browser,
      'linkBridge',
      'currentSelection',
    );
    expect(currentSelection).toMatchCustomSnapshot(testName);
  },
);

BrowserTestCase(
  'currentSelection when selection',
  // Safari has issues with key events
  { skip: skip.concat('safari', 'chrome') },
  async (client: any, testName: string) => {
    const browser = new Page(client);
    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await browser.type(editable, 'Normal Text');

    await browser.pressKey('ArrowLeft', 4);
    await browser.keys('Shift');
    await browser.pressKey('ArrowRight', 4);
    await browser.keys('Shift');

    const currentSelection = await getBridgeOutput(
      browser,
      'linkBridge',
      'currentSelection',
    );
    expect(currentSelection).toMatchCustomSnapshot(testName);
  },
);

BrowserTestCase(
  'currentSelection when cursor is on link',
  // Safari has issues with key events
  { skip: skip.concat('safari', 'chrome') },
  async (client: any, testName: string) => {
    const browser = new Page(client);
    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await browser.type(editable, '');
    await callNativeBridge(
      browser,
      'onLinkUpdate',
      'Atlassian',
      'https://www.google.com',
    );
    await browser.keys('ArrowLeft');

    const currentSelection = await getBridgeOutput(
      browser,
      'linkBridge',
      'currentSelection',
    );
    expect(currentSelection).toMatchCustomSnapshot(testName);
  },
);

// https://product-fabric.atlassian.net/browse/ED-6877
BrowserTestCase(
  'currentSelection when cursor is selecting a link',
  // Safari has issues with key events
  { skip: skip.concat('safari', 'chrome') },
  async (client: any, testName: string) => {
    const browser = new Page(client);
    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await browser.type(editable, '');
    await callNativeBridge(
      browser,
      'onLinkUpdate',
      'Atlassian',
      'https://www.google.com',
    );

    await browser.pressKey('ArrowLeft', 9);
    await browser.keys('Shift');
    await browser.pressKey('ArrowRight', 9);
    await browser.keys('Shift');

    const currentSelection = await getBridgeOutput(
      browser,
      'linkBridge',
      'currentSelection',
    );
    expect(currentSelection).toMatchCustomSnapshot(testName);
  },
);

BrowserTestCase(
  'currentSelection when cursor is selecting text and link',
  // Safari has issues with key events
  { skip: skip.concat('safari', 'chrome') },
  async (client: any, testName: string) => {
    const browser = new Page(client);
    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await browser.type(editable, 'This is a ');
    await callNativeBridge(
      browser,
      'onLinkUpdate',
      'link',
      'https://www.google.com',
    );

    await browser.pressKey('ArrowLeft', 10);
    await browser.keys('Shift');
    await browser.pressKey('ArrowRight', 10);
    await browser.keys('Shift');

    const currentSelection = await getBridgeOutput(
      browser,
      'linkBridge',
      'currentSelection',
    );

    expect(currentSelection).toMatchCustomSnapshot(testName);
  },
);
