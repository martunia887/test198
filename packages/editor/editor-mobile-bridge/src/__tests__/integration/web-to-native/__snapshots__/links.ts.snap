// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currentSelection when cursor is on link 1`] = `
Array [
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": 82,
    "left": 0,
    "right": 0,
    "text": "Atlassian",
    "top": 63,
    "url": "https://www.google.com",
  },
]
`;

exports[`currentSelection when cursor is selecting a link 1`] = `
Array [
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": 82,
    "left": 0,
    "right": 0,
    "text": "Atlassian",
    "top": 63,
    "url": "https://www.google.com",
  },
  Object {
    "bottom": 82,
    "left": 0,
    "right": 0,
    "text": "Atlassian",
    "top": 63,
    "url": "https://www.google.com",
  },
  Object {
    "bottom": 82,
    "left": 0,
    "right": 0,
    "text": "Atlassian",
    "top": 63,
    "url": "https://www.google.com",
  },
  Object {
    "bottom": 82,
    "left": 0,
    "right": 0,
    "text": "Atlassian",
    "top": 63,
    "url": "https://www.google.com",
  },
  Object {
    "bottom": 82,
    "left": 0,
    "right": 0,
    "text": "Atlassian",
    "top": 63,
    "url": "https://www.google.com",
  },
  Object {
    "bottom": 82,
    "left": 0,
    "right": 0,
    "text": "Atlassian",
    "top": 63,
    "url": "https://www.google.com",
  },
  Object {
    "bottom": 82,
    "left": 0,
    "right": 0,
    "text": "Atlassian",
    "top": 63,
    "url": "https://www.google.com",
  },
  Object {
    "bottom": 82,
    "left": 0,
    "right": 0,
    "text": "Atlassian",
    "top": 63,
    "url": "https://www.google.com",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": 82,
    "left": 0,
    "right": 0,
    "text": "Atlassian",
    "top": 63,
    "url": "https://www.google.com",
  },
]
`;

exports[`currentSelection when cursor is selecting text and link 1`] = `
Array [
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": 82,
    "left": 51.84375,
    "right": 51.84375,
    "text": "link",
    "top": 63,
    "url": "https://www.google.com",
  },
  Object {
    "bottom": 82,
    "left": 51.84375,
    "right": 51.84375,
    "text": "link",
    "top": 63,
    "url": "https://www.google.com",
  },
  Object {
    "bottom": 82,
    "left": 51.84375,
    "right": 51.84375,
    "text": "link",
    "top": 63,
    "url": "https://www.google.com",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": " ",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": " i",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": " is",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": " is ",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": " is a",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": " is a ",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
]
`;

exports[`currentSelection when no selection 1`] = `
Array [
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
]
`;

exports[`currentSelection when selection 1`] = `
Array [
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "T",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "Te",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "Tex",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "Text",
    "top": -1,
    "url": "",
  },
]
`;
